packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -195,8 +204,19 @@ export const useUpdate = <RecordType extends RaRecord = any, ErrorType = Error>(
                     'useUpdate mutation requires a non-empty data object'
                 );
             }
+            if (getMutateMiddlewares) {
+                const updateMiddleware = getMutateMiddlewares(
+                    dataProvider.update
```

Comment:
I think you should bind this function to the data provider (`dataProvider.update.bind(dataProvider)`), in case developers use `this` in their dataProvider.